fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructurePool.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import net.minecraft.structure.pool.StructurePool;
+import net.minecraft.structure.pool.StructurePoolElement;
+import net.minecraft.util.Identifier;
+
+/**
+ * Represents a modifiable structure pool that has several helper methods for modders.
+ */
+public interface FabricStructurePool {
```

Comment:
Maybe `@ApiStatus.NonExtendable`?

Reply:
Doesn't that mean it shouldn't be implemented?
modmuss requested that the class and impl be split to also include an interface in case someone wants to use it.

Reply:
This API interface + impl class pattern is usually used so that we can change the impl's structure without breaking anything (or even move it to another module). I don't think users gain anything in particular from implementing this itf, but I guess it can stay how it is without the anno.

Reply:
This is fine, I dont believe we use that on other similar interfaces.